src/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,173 @@
+// SegmentedControl
+
+.SegmentedControl {
+  display: inline-flex;
+  background-color: var(--color-segmented-control-bg);
+  // stylelint-disable-next-line primer/borders
+  border-radius: var(--primer-borderRadius-medium, $border-radius);
+  // stylelint-disable-next-line primer/box-shadow
+  box-shadow: var(--primer-borderInset-thin, inset 0 0 0 $border-width) var(--color-border-default);
+}
+
+// Button -----------------------------------------
+
+.SegmentedControl-button {
+  position: relative;
+  display: inline-flex;
+  height: var(--primer-control-medium-size, 32px);
+  // stylelint-disable-next-line primer/spacing
+  padding: 0 var(--primer-control-medium-paddingInline-normal, 12px);
+  // stylelint-disable-next-line primer/typography
+  font-size: var(--primer-text-body-size-medium, $body-font-size);
+  color: var(--color-fg-default);
+  background-color: transparent;
+  // stylelint-disable-next-line primer/borders
+  border: var(--primer-borderWidth-thin, $border-width) $border-style transparent;
+  // stylelint-disable-next-line primer/borders
+  border-radius: var(--primer-borderRadius-medium, $border-radius);
+  transition: border-color, background-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  align-items: center;
+  justify-content: center;
+  gap: var(--primer-control-medium-gap, $spacer-2);
+
+  &:hover {
+    background-color: var(--color-segmented-control-hover-bg);
+    transition-duration: 120ms;
+  }
+
+  &:active {
+    background-color: var(--color-segmented-control-active-bg);
+    border-color: var(--color-segmented-control-active-border);
+    transition-duration: 60ms;
+  }
+
+  // Pressed
+
+  // TODO: Change to aria-current or aria-selected?
+  &[aria-pressed='true'] {
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    background-color: var(--color-segmented-control-pressed-bg);
+    border-color: var(--color-segmented-control-pressed-border);
+    box-shadow: var(--color-shadow-medium);
```

Comment:
@simurai can you make sure the shadow being used in the selected state doesn't overflow outside of the element boundaries? 

@mperrotti also, in the React implementation it looks like there's no shadow being used.

Reply:
There's no shadow in the React implementation, but I think it would be a nice addition to show depth. I'll update my PR.